mongodbatlas/data_lakes.go

```diff
@@ -0,0 +1,236 @@
+	IAMAssumedRoleARN string `json:"iamAssumedRoleARN,omitempty"`
+	TestS3Bucket      string `json:"testS3Bucket,omitempty"`
+}
+
+type CloudProviderConfig struct {
+	AWSConfig AwsCloudProviderConfig `json:"aws,omitempty"`
+}
+
+type DataProcessRegion struct {
+	CloudProvider string `json:"cloudProvider,omitempty"`
+	Region        string `json:"region,omitempty"`
+}
+
+type DataLakeStore struct {
+	Name        string `json:"name,omitempty"`
+	Provider    string `json:"provider,omitempty"`
+	Region      string `json:"region,omitempty"`
+	Bucket      string `json:"bucket,omitempty"`
+	Prefix      string `json:"prefix,omitempty"`
+	Delimiter   string `json:"delimiter,omitempty"`
+	IncludeTags *bool  `json:"includeTags,omitempty"`
+}
+
+type DataLakeDataSource struct {
+	StoreName     string `json:"storeName,omitempty"`
+	DefaultFormat string `json:"defaultFormat,omitempty"`
+	Path          string `json:"path,omitempty"`
+}
+
+type DataLakeCollection struct {
+	Name        string               `json:"name,omitempty"`
+	DataSources []DataLakeDataSource `json:"dataSources,omitempty"`
+}
+
+type DataLakeDatabaseView struct {
+	Name     string `json:"name,omitempty"`
+	Source   string `json:"source,omitempty"`
+	Pipeline string `json:"pipeline,omitempty"`
+}
+
+type DataLakeDatabase struct {
+	Name        string                 `json:"name,omitempty"`
+	Collections []DataLakeCollection   `json:"collections,omitempty"`
+	Views       []DataLakeDatabaseView `json:"views,omitempty"`
+}
+
+type Storage struct {
+	Databases map[string]DataLakeDatabase `json:"databases,omitempty"`
+	Stores    []DataLakeStore             `json:"stores,omitempty"`
+}
+
+// DataLake represents a data lake.
+type DataLake struct {
+	CloudProviderConfig CloudProviderConfig `json:"cloudProviderConfig,omitempty"` // Configuration for the cloud service where Data Lake source data is stored.
+	DataProcessRegion   DataProcessRegion   `json:"dataProcessRegion,omitempty"`   // Cloud provider region which clients are routed to for data processing.
+	GroupID             string              `json:"groupId,omitempty"`             // Unique identifier for the project.
+	Hostnames           []string            `json:"hostnames,omitempty"`           // List of hostnames for the data lake.
+	Name                string              `json:"name,omitempty"`                // Name of the data lake.
+	State               string              `json:"state,omitempty"`               // Current state of the data lake.
+	Storage             Storage             `json:"storage,omitempty"`             // Configuration for each data store and its mapping to MongoDB collections / databases.
+}
+
+// DataLakeReqPathParameters represents all the possible parameters to make the request
+type DataLakeReqPathParameters struct {
+	GroupID string `json:"groupId,omitempty"` // The unique identifier of the project for the Atlas cluster.
+	Name    string `json:"name,omitempty"`    // The name of the Data Lake.
```

Comment:
[q] given this is used to construct the url do we need the json tags?
[aside] I'm not a fan of putting the URL params as a struct and I'd rather just have explicit params like `func Get(groupId, name string)` but this is not a blocker since I know we have a mix of both approaches around

Reply:
Yeah we can do without these -- for the struct yeah I wasn't sure if we have a preferred approach so I can go with the other approach if thats slightly preferred

Reply:
@robcarlan-mlab up to you if you want to refactor, also consider that some of the functions that use this struct only check for the group and not the name but like I mentioned we have a bit of a mix on both so I won't block on this if you prefer to keep it

Reply:
I just refactored it :)